balances-tx-payment/rpc/src/lib.rs

```diff
@@ -0,0 +1,186 @@
+		+ Decode
+		+ Send
+		+ Sync
+		+ PartialEq
+		+ Into<NumberOrHex>
+		+ MaybeDisplay
+		+ Copy,
+	Block: BlockT,
+	C: ProvideRuntimeApi<Block> + HeaderBackend<Block> + Send + Sync + 'static,
+	C::Api: BalancesTxPaymentApiRuntimeApi<Block, Balance, AssetId, AssetBalance>,
+	C::Api: TransactionPaymentRuntimeApi<Block, Balance>,
+	Balance:
+		Codec + MaybeDisplay + Copy + TryInto<NumberOrHex> + Send + Sync + 'static + From<u128>,
+{
+	fn query_asset_fee_details(
+		&self,
+		asset_id: AssetId,
+		encoded_xt: Bytes,
+		at: Option<Block::Hash>,
+	) -> RpcResult<FeeDetails<NumberOrHex>> {
+		let api = self.client.runtime_api();
+		let at = BlockId::hash(at.unwrap_or_else(|| self.client.info().best_hash));
+
+		let encoded_len = encoded_xt.len() as u32;
+
+		let uxt: Block::Extrinsic = Decode::decode(&mut &*encoded_xt).map_err(|e| {
+			CallError::Custom(ErrorObject::owned(
+				Error::DecodeError.into(),
+				"Unable to query fee details.",
+				Some(format!("{:?}", e)),
+			))
+		})?;
+		let fee_details = api.query_fee_details(&at, uxt, encoded_len).map_err(|e| {
+			CallError::Custom(ErrorObject::owned(
+				Error::RuntimeError.into(),
+				"Unable to query fee details.",
+				Some(e.to_string()),
+			))
+		})?;
+
+		let try_into_rpc_balance = |value: AssetBalance| {
+			value.try_into().map_err(|_| {
+				JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+					ErrorCode::InvalidParams.code(),
+					format!("{} doesn't fit in NumberOrHex representation", value),
+					None::<()>,
+				)))
+			})
+		};
+
+		Ok(FeeDetails {
+			inclusion_fee: if let Some(inclusion_fee) = fee_details.inclusion_fee {
+				let base_fee = api
+					.balance_to_asset_balance(&at, inclusion_fee.base_fee, asset_id)
+					.map_err(|e| {
+						CallError::Custom(ErrorObject::owned(
+							Error::RuntimeError.into(),
+							"Unable to query balance conversion.",
+							Some(e.to_string()),
+						))
+					})?
+					.map_err(|_e| {
+						CallError::Custom(ErrorObject::owned(
+							Error::RuntimeError.into(),
+							"Unable to query balance conversion.",
+							Some("Unable to query balance conversion."),
+						))
+					})?;
+
+				let len_fee = api
+					.balance_to_asset_balance(&at, inclusion_fee.len_fee, asset_id)
+					.map_err(|e| {
+						CallError::Custom(ErrorObject::owned(
+							Error::RuntimeError.into(),
+							"Unable to query fee details.",
+							Some(e.to_string()),
+						))
+					})?
+					.map_err(|_e| {
+						CallError::Custom(ErrorObject::owned(
+							Error::RuntimeError.into(),
+							"Unable to query balance conversion.",
+							Some("Unable to query balance conversion."),
+						))
+					})?;
+
+				let adjusted_weight_fee = api
+					.balance_to_asset_balance(&at, inclusion_fee.adjusted_weight_fee, asset_id)
+					.map_err(|e| {
+						CallError::Custom(ErrorObject::owned(
+							Error::RuntimeError.into(),
+							"Unable to query fee details.",
+							Some(e.to_string()),
+						))
+					})?
+					.map_err(|_e| {
```

Comment:
Why are you dropping this error?

Reply:
...which is weird because `thiserror` crate claims that

A Display impl is generated for your error if you provide #[error("...")] messages on the struct or each variant of your enum, as shown above in the example.

which we actually do